api/src/main/java/io/opencensus/trace/SpanBuilder.java

```diff
@@ -223,6 +224,98 @@ public final Scope startScopedSpan() {
     return new ScopedSpanHandle(startSpan());
   }
 
+  /**
+   * Wraps a {@link Runnable} so that it executes with the {@code Span} as the current {@code Span}
+   * and saves the rest of the current Context, and ends the {@code Span} after the {@code Runnable}
+   * is run.
+   *
+   * <p>This method saves the entire {@link io.grpc.Context} when wraps the {@code Runnable}.
+   *
+   * <p>This is equivalent with following code:
+   *
+   * <pre><code>
+   * Span span = tracer.spanBuilder("MyRunnableSpan").startSpan();
+   * tracer.wrap(span, new Runnable() {
+   *    {@literal @}Override
+   *     public void run() {
+   *       try {
+   *         myRunnable.run();
+   *       } finally {
+   *         span.end();
+   *       }
+   *     }
+   * });
+   * </code></pre>
+   *
+   * @param runnable the {@code Runnable} to run in the {@code Span}.
+   */
+  public final Runnable startSpanAndWrap(final Runnable runnable) {
```

Comment:
maybe recordInSpan? or decorate or wrap?

ex. spanBuilder.decorate(runnable), spanBuilder.wrap(runnable) makes sense to me. wrap makes most sense sense, if we start the span deferred.

Reply:
@sebright said

"I think "wrap" is a little unclear. One suggestion is "startSpanInRunnable"."

Currently we have options:
startSpanAndWrap
startSpanInRunnable/startSpanInCallable
wrap
decorate

I personally like the first one, but I would like to hear other opinions.

Reply:
I suggested startSpanInRunnable when I had assumed that startSpan was called inside the runnable, so I don't think it's the best choice now.  I think that startSpanAndWrap is probably the most descriptive, though the word "wrap" sounds too general.  I'll comment again if I think of anything else.